### DISCUSSION TITLE
Suggestion: Make the API isomorphic

### DISCUSSION THREAD

**Discussion by osseonews** [swellstores/community#137] [Ideas]
Just a suggestion, you should make the API Backend isomorphic, so it can be used in edge functions with NextJs or with Cloudflare Workers. Over time, edge will become standard, as it is much faster.  Right now, the only way to use backend in Edge is to basically use the curl version of the API and pass in the variables with fetch. it's quite tedious.

**Comment by maarten2424** [comment]
Hey @osseonews i'm assuming you want to use the Frontend-API on on edge functions. We are currently working on a more abstract integration of the Frontend-API, which will be backend-isomorphic and can be used wherever! You can keep track of the progress [here](https://github.com/swellstores/swell-sdk)

**Reply by osseonews** [reply]
Yes, exactly but not just the frontend api, the backend also. There are things you can do with backend that you can't do on frontend.

**Reply by ericingram** [reply]
We'd like to fix that in the future by consolidating them into one API with improved frontend/backend permissions, but we are still working on a new API versioning strategy as a pre-requisite.
